addons/sale_management/models/sale_order.py

```diff
@@ -185,7 +185,7 @@ def product_id_change(self):
         if self.product_id and self.order_id.sale_order_template_id:
             for line in self.order_id.sale_order_template_id.sale_order_template_line_ids:
                 if line.product_id == self.product_id:
-                    self.name = line.with_context(lang=self.order_id.partner_id.lang).name + self._get_sale_order_line_multiline_description_variants()
+                    self.name = line.with_context(lang=self.order_id.partner_id.lang).name + self.with_context(lang=self.order_id.partner_id.lang)._get_sale_order_line_multiline_description_variants()
```

Comment:
Minor request to keep the line length reasonable (should have said it before, mb).
Would be cleaner to extract the lang spec in a separate line before the use in the `with_context`
```python
lang = self.order_id.partner_id.lang
self.name = line.with_context(lang=lang).name + self.with_context(lang=lang). ...
```